benchmark/benchmark.sh

```diff
@@ -103,49 +98,102 @@ stop_all() {
 }
 
 set_latency() {
-    docker exec $1 tc qdisc add dev eth0 root handle 1: prio bands 7
-    docker exec $1 tc qdisc add dev eth0 parent 1:4 handle 40: netem delay $5
-    docker exec $1 tc qdisc add dev eth0 parent 1:5 handle 50: netem delay $6
-    docker exec $1 tc qdisc add dev eth0 parent 1:6 handle 60: netem delay $7
-    docker exec $1 tc filter add dev eth0 protocol ip parent 1:0 u32 match ip dst $2 flowid 1:4
-    docker exec $1 tc filter add dev eth0 protocol ip parent 1:0 u32 match ip dst $3 flowid 1:5
-    docker exec $1 tc filter add dev eth0 protocol ip parent 1:0 u32 match ip dst $4 flowid 1:6
+    docker exec ${1} tc filter add dev eth0 protocol ip parent 1:0 u32 match ip dst ${2} flowid 1:${4}
+    docker exec ${1} tc qdisc add dev eth0 parent 1:${4} handle ${4}0: netem delay ${3}
 }
 
 set_cluster_latency() {
-    set_latency node0 ${SERVERS[1]} ${SERVERS[2]} ${CLIENT} 50ms 50ms 75ms &
-    set_latency node1 ${SERVERS[0]} ${SERVERS[2]} ${CLIENT} 50ms 50ms 75ms &
-    set_latency node2 ${SERVERS[0]} ${SERVERS[1]} ${CLIENT} 50ms 50ms 25ms &
-    set_latency client ${SERVERS[0]} ${SERVERS[1]} ${SERVERS[2]} 75ms 75ms 25ms &
+    cluster_size=$1
+    docker exec client tc qdisc add dev eth0 root handle 1: prio bands $((cluster_size + 4))
+    for ((j = 1; j < ${cluster_size}; j++)); do
+        set_latency client 172.20.0.$((j + 2)) 75ms $((j + 3)) &
+    done
+    set_latency client 172.20.0.$((cluster_size + 2)) 50ms $((cluster_size + 3)) &
+    for ((i = 1; i <= ${cluster_size}; i++)); do
+        docker exec node$i tc qdisc add dev eth0 root handle 1: prio bands $((cluster_size + 4))
+        idx=4
+        for ((j = 1; j <= ${cluster_size}; j++)); do
+            if [ ${i} -ne ${j} ]; then
+                set_latency node$i 172.20.0.$((j + 2)) 50ms ${idx} &
+                idx=$((idx + 1))
+            fi
+        done
+        if [[ ${i} -eq ${cluster_size} ]]; then
+            set_latency node${i} 172.20.0.2 50ms ${idx} &
+        else
+            set_latency node${i} 172.20.0.2 75ms ${idx} &
+        fi
+    done
     wait
 }
 
+run_container() {
```

Comment:
I recommend to add some comment before the complex functions to tell the meaning of each arguments, as it's hard to tell in a shell script.